### PR TITLE
Display error component

### PR DESCRIPTION
**Pull request by Wilker0725**
**Changes**
Description of changes...

- [delete] branch

**Changelog entry**

- [WARNING] Description of change in `ErrorDisplay Component`

**Additional information**

- You may refer to this [ticket](https://www.figma.com/file/gKejDx4Eyihzw0mroc3n1h/LifeSG-Illustration-Library?node-id=2819-58094&t=4vATRzph7cACP2hb-0)
- Any other information you would like to include

### CODE REVIEW

**Comment by Wilker0725** [comment]
| Legacy Component            | New / Updated Component     | Updated |
| --------------------------- | --------------------------- | ------- |
| 400 Error                   | 400 Error                   | Yes     |
| 403 Error                   | 403 Error                   | Yes     |
| 404 Error                   | 404 Error                   | Yes     |
| 408 Error                   | 408 Error                   | Yes     |
| 500 Error                   | 500 Error                   | Yes     |
| 503 Error                   | 503 Error                   | Yes     |
|                             | 504 Error                   | Yes     |
|                             | Confirmation modal          | Yes     |
|                             | Inactive state              | Yes     |
|                             | Insufficient credits        | Yes     |
|                             | Link error state            | Yes     |
|                             | Logged-out                  | Yes     |
|                             | Warning                     | Yes     |
| Maintenance                 |                             | No      |
|                             | No Item Found               | Yes     |
|                             | Payment unsuccessful        | Yes     |
|                             | Transfer unsuccessful       | Yes     |
| Unsupported Browser         | Unsupported Browser         | Yes     |
| Partially Supported Browser | Partially Supported Browser | Yes     |

File: tests/error-display/error-display.spec.tsx [hunk 1]
```diff
@@ -62,173 +65,67 @@ describe("ErrorDisplay", () => {
...
+                ERROR_DISPLAY_DATA.get("unsupported-browser"),
+            ],
+            [
+                "partially-supported-browser",
+                ERROR_DISPLAY_DATA.get("partially-supported-browser"),
+            ],
+        ];
+
+        test.each(testData)(
+            "renders %s error correctly",
```
Comment (qroll) [inline tests/error-display/error-display.spec.tsx:108]:
```suggestion
            "should render %s error correctly",
```

File: tests/error-display/error-display.spec.tsx [hunk 2]
```diff
@@ -62,173 +65,67 @@ describe("ErrorDisplay", () => {
...
-            expect(screen.getByRole("img")).toHaveAttribute(
-                "src",
-                error.img.src
-            );
-            expect(
-                screen.getByTestId(ERROR_DESCRIPTION_TEST_ID).textContent
-            ).toBe(error.description);
-        });
+    describe("ErrorDisplay components", () => {
+        const testData = [
```
Comment (qroll) [inline tests/error-display/error-display.spec.tsx:69]: should be able to get error from `ERROR_DISPLAY_DATA.get(type)` so there's no need to specify it in `testData`

then `testData` can be declared as `ErrorDisplayType[]`

**Review by qroll — CHANGES_REQUESTED** [review]

File: tests/error-display/error-display.spec.tsx (hunk 2 above)
Comment (Wilker0725) [reply tests/error-display/error-display.spec.tsx:69]: arh.. my bad. ok!

**Review by qroll — APPROVED** [review]

**Pull request merged**